sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdk.java

```diff
@@ -51,10 +51,10 @@ static AutoConfiguredOpenTelemetrySdk create(
   public abstract OpenTelemetrySdk getOpenTelemetrySdk();
 
   /** Returns the {@link Resource} that was auto-configured. */
-  public abstract Resource getResource();
+  abstract Resource getResource();
```

Comment:
Doesn't this remove the only supported way to get an autoconfigured Resource? In cases where people want to manually configured everything except the Resource, I believe this was the only possible way to do it. This is a useful thing to be able to do, and this change makes it even harder to access.  

I've been a strong proponent of having a way to get access to an autoconfigured Resource instance, and have had to settle for this being the only way. This change would now make it impossible to do in a supported way.

If we're going to remove this, we should provide a supported way for users to get the resource.

Reply:
> Doesn't this remove the only supported way to get an autoconfigured Resource?

Well its been moved to an internal API, but yes, it removes it from the proposed stable API. 

Can you remind me of some of the use cases of accessing the autoconfigured resource? My motivation for this is to be conservative about the public API surface area, and I struggled to think of reasons why you would want to access the resource.

Reply:
My use-case is: I don't want to use the full auto-configure mechanisms because getting it right for my use-case is more trouble than just writing a few lines of java code to do it. But, I _do_ want the auto-configured resource, since that's pretty simple.

Reply:
@jkwatson this is related to #5238?

Reply:
Having the resource accessible was the entire point behind the [`AgentListener`](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/AgentListener.java) accepting an `AutoConfiguredOpenTelemetrySdk` instance. We use it in a couple of places in our distro (e.g. [profiling](https://github.com/signalfx/splunk-otel-java/blob/6ded9e025087e723efb8163b986eac5aba435189/profiler/src/main/java/com/splunk/opentelemetry/profiler/JfrActivator.java#L70))

(`getConfig()` is also used in several places, including the vanilla OTel javaagent, but we can just use the internal API there)

Reply:
@mateuszrzeszutek can that profiling instrumentation use LoggerProvider / TracerProvider / EventEmitterProvider/ MeterProvider instead?

If we're going to have a public `getResource` method for obtaining the autoconfigured resource, what will our reasoning be for not exposing accessors for the autoconfigured components which comprise the providers? E.g. spanlimits, views, exporters, processors, samplers. I'm sure we'll receive these types of requests (I vaguely remember someone already asking for an accessor for an autoconfigured OTLP exporter).. should we deny them? If so, what's different about resource?

Reply:
> @mateuszrzeszutek can that profiling instrumentation use LoggerProvider / TracerProvider / EventEmitterProvider/ MeterProvider instead?

I think we could make that work; currently the exporter we're using for profiling sends an extra header that allows differentiating profiling data from normal logs. We probably could use the `AutoConfigurationCustomizer` to split the logs/profiler streams into different exporters.

I looked through the other usages of the resource, and the only other one that really needs the resource is [this class that enhances the legacy logging context injection mechanism](https://github.com/signalfx/splunk-otel-java/blob/6ded9e025087e723efb8163b986eac5aba435189/custom/src/main/java/com/splunk/opentelemetry/resource/ResourceAttributesToSystemProperties.java#L32).

> If we're going to have a public `getResource` method for obtaining the autoconfigured resource, what will our reasoning be for not exposing accessors for the autoconfigured components which comprise the providers? E.g. spanlimits, views, exporters, processors, samplers. I'm sure we'll receive these types of requests (I vaguely remember someone already asking for an accessor for an autoconfigured OTLP exporter).. should we deny them? If so, what's different about resource?

IMO resource is slightly different because it describes the instrumented application, not the settings of the SDK. Resource is a piece of telemetry; just as resource providers are instrumentation.

Reply:
Now that #5554 is merged and there is a public API for obtaining the resource configured from environment variables / system properties, I've updated this PR to remove even the experimental accessor to get the autoconfigured resource.

With the new public API, I feel fairly strongly that any code accessing `AutoConfiguredOpenTelemetrySdk.getResource()` shouldn't.

Reply:
Resource configured from env vars != the full autoconfigured resource

I'm quite certain that we will continue to use the `AutoConfiguredOpenTelemetrySdk#getResource()` method to get the full resource; we'll just have to move the reflection logic to our distro.

Reply:
Anyway, I don't want to block this issue on one vendor's complaints (some of which might disappear in the future, when we have a profiling signal and SDK); even more when we have a workaround for that.

Reply:
> Resource configured from env vars != the full autoconfigured resource

Correct. I'm torn on this. I'm ok with re-adding an internal method to continue accessing the full autoconfigured resource, but feel that long term, folks should work exclusively through the providers. I worry that keeping an internal method gives hope that someday accessing the full autoconfigured resource will become a public API, where in actuality I think its more likely to go away altogether.

Reply:
> feel that long term, folks should work exclusively through the providers.

Yeah, I totally agree with that. Long-term (or maybe even mid-term, once we have the profiling SDK) our (Splunk's) use cases will go away; and with #5554 implemented SDK users themselves should have no reason for that `getResource()` method either.
I'm in favor of merging this PR as it is -- there are workaround that we can use in the Splunk distro; and the cause (stable autoconfigure module) is important enough.